accounts/url_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package accounts
+
+import (
+	"testing"
+)
+
+func TestParseURL(t *testing.T) {
+	url, err := parseURL("https://ethereum.org")
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if url.Scheme != "https" {
+		t.Errorf("expected: %v, got: %v", "https", url.Scheme)
+	}
+	if url.Path != "ethereum.org" {
+		t.Errorf("expected: %v, got: %v", "ethereum.org", url.Path)
+	}
+
+	_, err = parseURL("ethereum.org")
+	if err == nil {
+		t.Error("expected err, got: nil")
+	}
+}
+
+func TestURLString(t *testing.T) {
+	url := URL{Scheme: "https", Path: "ethereum.org"}
+	if url.String() != "https://ethereum.org" {
+		t.Errorf("expected: %v, got: %v", "https://ethereum.org", url.String())
+	}
+
+	url = URL{Scheme: "", Path: "ethereum.org"}
+	if url.String() != "ethereum.org" {
+		t.Errorf("expected: %v, got: %v", "ethereum.org", url.String())
+	}
+}
+
+func TestTerminalString(t *testing.T) {
+	url := URL{Scheme: "https", Path: "ethereum.org"}
+	if url.TerminalString() != "https://ethereum.org" {
+		t.Errorf("expected: %v, got: %v", "https://ethereum.org", url.TerminalString())
+	}
+
+	url = URL{Scheme: "https", Path: "ethereum.org/aaaaaaaaaaa"}
+	if url.TerminalString() != "https://ethereum.org/aaaaaaaaaaa" {
+		t.Errorf("expected: %v, got: %v", "https://ethereum.org/aaaaaaaaaaa", url.TerminalString())
+	}
+
+	url = URL{Scheme: "https", Path: "ethereum.org/aaaaaaaaaaab"}
+	if url.TerminalString() != "https://ethereum.org/aaaaaaaaaa…" {
+		t.Errorf("expected: %v, got: %v", "https://ethereum.org/aaaaaaaaaa…", url.TerminalString())
+	}
+}
```

Comment:
The `TerminalString` is just a helper method to make URLs a bit shorter when printing them in the logs. This method may change if we adjust things in the logger, so I'd rather not have a test for it. It's not an API, just some pretty printing thing, it would just be bothersome to keep updating tests when adjusting log formats.